### PR TITLE
have separate intercom secret in env var and return the hash based on the client

### PR DESCRIPTION
**Pull request by sywhb**

### CODE REVIEW

File: packages/api/src/util.ts [hunk 1]
```diff
@@ -268,6 +274,9 @@ export function getEnv(): BackendEnv {
   const intercom = {
     token: parse('INTERCOM_TOKEN'),
     secretKey: parse('INTERCOM_SECRET_KEY'),
+    webSecret: parse('INTERCOM_WEB_SECRET'),
```
Comment (jacksonh) [inline packages/api/src/util.ts:277]: Are these ones mandatory by default?

Comment (sywhb) [reply packages/api/src/util.ts:277]: No

Comment (sywhb) [reply packages/api/src/util.ts:277]: Should we make them mandatory by default? @jacksonh

Comment (jacksonh) [reply packages/api/src/util.ts:277]: @sywhb no, i think they should be optional so it doesn't break for self hosting. I just wanted to confirm.

**Review by jacksonh — APPROVED** [review]

**Pull request merged**